### PR TITLE
Update to PureScript v0.15.0

### PR DESCRIPTION
**Pull request by sigma-andex**
**Description of the change**

Backlinking to purescript/purescript#4244

Migrates FFI to ES modules

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by sigma-andex** [comment]
🏓 @thomashoneyman @JordanMartinez

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Looks good, but it seems we're missing a changelog entry.

**Comment by JordanMartinez** [comment]
CI is failing due to unused imports.

File: src/Node/FS/Async.purs [hunk 1]
```diff
@@ -64,42 +64,41 @@ foreign import handleCallbackImpl ::
...
+foreign import readdir_ :: Fn2 FilePath (JSCallback (Array FilePath)) Unit
+foreign import utimes_ :: Fn4 FilePath Int Int (JSCallback Unit) Unit
+foreign import readFile_ :: forall a opts. Fn3 FilePath { | opts } (JSCallback a) Unit
+foreign import writeFile_ :: forall a opts. Fn4 FilePath a { | opts } (JSCallback Unit) Unit
+foreign import appendFile_ :: forall a opts. Fn4 FilePath a { | opts } (JSCallback Unit) Unit
+foreign import exists_ :: forall a. Fn2 FilePath (Boolean -> a) Unit
+foreign import open_ :: Fn4 FilePath String (Nullable FileMode) (JSCallback FileDescriptor) Unit
+foreign import read_ :: Fn6 FileDescriptor Buffer BufferOffset BufferLength (Nullable FilePosition) (JSCallback ByteCount) Unit
+foreign import write_ :: Fn6 FileDescriptor Buffer BufferOffset BufferLength (Nullable FilePosition) (JSCallback ByteCount) Unit
+foreign import close_ :: Fn2 FileDescriptor (JSCallback Unit) Unit
```
Comment (thomashoneyman) [inline src/Node/FS/Async.purs:92]: We can keep this as-is, but typically we'd name this

```purs
foreign import closeImpl :: ...
```

where `*Impl` denotes an FFI implementation

Comment (thomashoneyman) [reply src/Node/FS/Async.purs:92]: Admittedly, this library already diverged from our typical best practices with this big `fs` record 😆

**Review by thomashoneyman — APPROVED** [review]

**Review by JordanMartinez — APPROVED** [review]

**Pull request merged**